### DISCUSSION TITLE
[UI] Some suggestions and features.

### DISCUSSION THREAD

**Discussion by malnaanah** [olive-editor/olive#2087] [Ideas]
**Commit Hash** 
c06d17cf

**Platform** 
Kubuntu 22.04

**Summary**
I have been trying Olive 0.2 for a while, and it has been more simplified and usable then before, and I believe it has great potentials and capabilities. Here I have some suggestions, features, and notes about it.

1- Saving changes to UI, such as locking panels, direction of nodes, project bin list type.
2- Adding ability to save custom UI design.
3- Adding more math nodes.
4- Adding effect strip [very important], this could be implemented by enabling the empty strip to take input from video tracks so that the effects can be applied easily in a non-destructive manner to any part of videos.
5- Improving transition strip to include more transitions, especially wiping transition (like Kdenlive) with the ability to load custom wipe images.
6- Ability to create custom effects by grouping nodes together (there is an option to group nodes, but olive crashes after the name of the group is entered.
7- Removing the constrains on UI panel size, especially the timeline, because it is taking large unnecessary vertical space.
8- adding option to show only the properties of the selected node.
9- Changing the default UI design to give more space for node editing, and changing the direction of nodes to Bottom-to-top. I have attached image for the suggested UI design. 

**Additional Information / Output**
![Screenshot_20221103_102058](https://user-images.githubusercontent.com/45626803/199667510-74e52d23-d44b-413e-bb6e-fa0f94f6efc9.png)

**Comment by malnaanah** [comment]
Some of the suggested additional math functions:
1- Absolute value.
2- Limit function: upper, lower, and both (level).
3- exponential function.

Additional suggestions:
1- Distinguish math inputs as value1 and value2.
2- Not compressing the appimage files, because they are already compressed.

**Reply by mcDandy** [reply]
It Is possible to make abs in nodes. It is sqrt(input*input)) which is unnecessery slow. Exponential function could be approximated using Taylor polynom (or pow node with e as the first parameter).

These things however might be slow or unwieldy So I am in favor of adding them.

**Comment by malnaanah** [comment]
Another notice:
The position and size node input, for transform and shape nodes, need to to be split into x and y coordinates.
